seqr/views/apis/individual_api.py

```diff
@@ -787,19 +787,48 @@ def save_individuals_metadata_table_handler(request, project_guid, upload_file_i
 
 @login_and_policies_required
 def get_individual_rna_seq_data(request, individual_guid):
+    expr_data, expr_genes = _get_rna_seq_data(RnaSeqOutlier, request, individual_guid)
+    splice_data, splice_genes = _get_rna_seq_data(RnaSeqSpliceOutlier, request, individual_guid)
+
+    expr_genes.update(splice_genes)
+    data = {'outliers': expr_data} if expr_data else {}
+    data.update({'spliceOutliers': splice_data} if expr_data else {})
+    return create_json_response({
+        'rnaSeqData': {individual_guid: data},
+        'genesById': expr_genes,
+        'familiesByGuid': {
+            family['familyGuid']: family for family in _get_json_for_families(
+                Family.objects.filter(pk=Individual.objects.get(guid=individual_guid).family.pk), request.user)
+        },
+        'igvSamplesByGuid': {
+            igvs['sampleGuid']: igvs for igvs in get_json_for_samples(IgvSample.objects.filter(individual__guid=individual_guid))
+        },
+        'individualsByGuid': {
+            indiv['individualGuid']: indiv for indiv in _get_json_for_individuals(
+                Individual.objects.filter(guid=individual_guid)
+            )
+        },
+    })
+
+
+def _get_rna_seq_data(model_cls, request, individual_guid):
     individual = Individual.objects.get(guid=individual_guid)
     check_project_permissions(individual.family.project, request.user)
-    outlier_data = RnaSeqOutlier.objects.filter(sample__individual=individual, sample__is_active=True)
+    outlier_data = model_cls.objects.filter(sample__individual=individual, sample__is_active=True).prefetch_related('sample')
 
     rna_seq_data = {
-        data['geneId']: data for data in get_json_for_rna_seq_outliers(outlier_data)
+        data['geneId']: data for data in get_json_for_rna_seq_outliers(
+            outlier_data,
+            nested_fields=[
+                {'fields': ('sample', 'sample_id'), 'key': 'sampleId'},
+                {'fields': ('sample', 'tissue_type'), 'key': 'tissueType'}
```

Comment:
We just need one `sampleId` and `tissueType` but we replicate then to all the data rows. Are there any better ways to do it?

Reply:
Weirdly, when I call `getSamplesByGuid` in the new RnaSeqResultPage, it returns an empty `samplesByGuid`. Same for the `individualsByGuid` and `familiesByGuid`.

Reply:
I think switching the routing the way I suggest so the family data is all loaded will fix this issue